src/spec.ts

```diff
@@ -128,7 +148,7 @@ export function isSomeUnitSpec(spec: ExtendedSpec): spec is ExtendedUnitSpec | U
   return spec['mark'] !== undefined;
 }
 
-export function isLayerSpec(spec: ExtendedSpec): spec is LayerSpec {
+export function isLayerSpec(spec: ExtendedSpec | ExtendedFacetSpec): spec is LayerSpec {
```

Comment:
I tried to add `ExtendedFacetSpec` to `ExtendedSpec` but that will break the spec `normalize()` function because they are not expecting `ExtendedFacetSpec`...